app/javascript/history/history.jsx

```diff
@@ -1,213 +1,106 @@
 import { h, Component } from 'preact';
 import { PropTypes } from 'preact-compat';
 import debounce from 'lodash.debounce';
-import setupAlgoliaIndex from '../src/utils/algolia';
+
+import {
+  defaultState,
+  loadNextPage,
+  onSearchBoxType,
+  performInitialSearch,
+  search,
+  toggleTag,
+} from '../searchableItemList/searchableItemList';
```

Comment:
I've extracted a few functions that are common between the two components. 

The pro composition approach is my favorite, and also recommended by React itself:

> If you want to reuse non-UI functionality between components, we suggest extracting it into a separate JavaScript module. The components may import it and use that function, object, or a class, without extending it.

from https://reactjs.org/docs/composition-vs-inheritance.html#so-what-about-inheritance